snuba/query/snql/__init__.py

```diff
@@ -151,6 +213,63 @@ def visit_quoted_literal(
     ) -> Literal:
         return visit_quoted_literal(node, visited_children)
 
+    def visit_where_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Expression, Any]
+    ) -> Expression:
+        _, _, conditions, _ = visited_children
+        return conditions
+
+    def visit_having_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Expression, Any]
+    ) -> Expression:
+        _, _, conditions, _ = visited_children
+        return conditions
+
+    def visit_and_expression(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any, Expression]
+    ) -> Expression:
+        _, left_condition, _, and_condition = visited_children
+
+        # in the case of one Condition
+        # and_condition will be an empty Node
+        if isinstance(and_condition, Node):
+            return left_condition
+        if isinstance(and_condition, (AndTuple, OrTuple)):
+            _, exp = and_condition
+            return combine_and_conditions([left_condition, exp])
+        elif isinstance(and_condition, list):
+            for elem in and_condition:
+                _, exp = elem
+                left_condition = combine_and_conditions([left_condition, exp])
```

Comment:
It feels unpythonic to call `combine_and_conditions` once for every element in the list, particularly since that function takes a list as an argument. I would build the list first, and then pass the whole thing into `combine_and_conditions`.

```
args = [left_condition] + [elem[1] for elem in and_condition]
combine_and_conditions(args)
```

Reply:
Good point - I changed the logic in my code so that I'm building the list first and then passing it once to combine_and_conditions / combine_or_conditions. Thank you!